### PR TITLE
fix: polish desktop task detail sidebar

### PR DESCRIPTION
**Pull request by Nek-12**
## Summary
- Paginate and virtualize task detail comments/activity in one scroll surface
- Align task-detail islands, tabs, comment cards, title/comment fields, spacing, and sidebar radii with the island design system
- Add typed destination sidebar sizing defaults and remove dead agent-browser proof scripts

## Verification
- pnpm --dir apps/desktop exec vitest run src/app/sidebarDestinationSizing.test.ts src/app/sidebarSizing.test.ts src/app/sidebar.test.tsx
- pnpm --dir apps/desktop exec vitest run src/features/task-detail/TaskDetailDialog.test.tsx
- pnpm --dir apps lint
- pnpm --dir apps typecheck
- pnpm --dir apps build

## Summary by CodeRabbit

* **New Features**
  * Added task comment pagination and listing capabilities
  * Implemented sidebar sizing preferences for different task destinations
  * Added informational prompt placeholders in workflow editor

* **Improvements**
  * Updated workflow editor to use "transition" terminology
  * Enhanced task detail layout with improved component organization
  * Refactored sleep prevention with automatic process restart capability
  * Updated CLI commands with pagination support for tasks and workflows
  * Refined UI typography and color scheme

### CODE REVIEW

File: cli/builder/task_command.go [hunk 1]
```diff
@@ -133,40 +182,91 @@ func taskStartSubcommand(args []string, stdout io.Writer, stderr io.Writer) int
 		fmt.Fprintln(stderr, err)
 		return 1
 	}
-	fmt.Fprintf(stdout, "run_id\t%s\nplacement_id\t%s\ntransition_id\t%s\n", resp.RunID, resp.PlacementID, resp.TransitionID)
+	detail, err := waitForWorkflowTaskRunSession(context.Background(), remote, taskID, resp.RunID, taskStartSessionPollTimeout, taskStartSessionPollInterval)
+	if err != nil {
+		fmt.Fprintln(stderr, err)
+		return 1
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/task_command.go:188]: By design: a follow-up read/poll failure after the mutation keeps exit 1 so failures are never silently swallowed. The task-start path intentionally waits briefly for a session id and surfaces a non-zero exit if it doesn't attach. Keeping current behavior per maintainer decision.

File: cli/builder/task_command.go [hunk 2]
```diff
@@ -96,7 +140,12 @@ func taskCreateSubcommand(args []string, stdout io.Writer, stderr io.Writer) int
 		fmt.Fprintln(stderr, err)
 		return 1
 	}
-	fmt.Fprintf(stdout, "task_id\t%s\nshort_id\t%s\nworkflow_id\t%s\n", resp.Task.ID, resp.Task.ShortID, resp.Task.WorkflowID)
+	task, err := getWorkflowTaskByID(context.Background(), remote, resp.Task.ID)
+	if err != nil {
+		fmt.Fprintf(stderr, "created task %s but failed to load task detail for output: %v\n", resp.Task.ID, err)
+		return 1
+	}
+	writeTaskDetail(stdout, task)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/task_command.go:148]: Intentional: these commands return exit 1 when the follow-up read/poll after the mutation fails, so a failed verification is never reported as success. Maintainer decision is to keep the strict exit code.

File: server/workflowstore/store.go [hunk 3]
```diff
@@ -430,43 +444,59 @@ func (s *Store) ListWorkflows(ctx context.Context, req ListWorkflowsRequest) (Li
...
+		where = "WHERE " + strings.Join(clauses, " AND ")
+	}
+	cursorClause := ""
+	if cursor.hasValue {
+		cursorClause = "WHERE activity_at_unix_ms < ? OR (activity_at_unix_ms = ? AND id < ?)"
+		args = append(args, cursor.activityAtUnixMs, cursor.activityAtUnixMs, cursor.workflowID)
+	}
+	args = append(args, pageSize+1)
+	queryText := strings.Replace(strings.TrimSuffix(listWorkflowsQuery, "\n"), "{{clause}}", where, 1)
+	queryText = strings.Replace(queryText, "{{cursor_clause}}", cursorClause, 1)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowstore/store.go:475]: Not a bug: the two placeholders live in different query scopes. `{{clause}}` is injected inside the `workflow_list` CTE (`FROM workflows {{clause}}`) and `{{cursor_clause}}` in the outer query (`FROM workflow_list {{cursor_clause}}`), so they never combine into a double WHERE. The suggested `AND` prefix would actually break it (`FROM workflow_list AND ...`). Added a regression test covering filtered+paginated listing across pages (commit 1e56e5f0).

File: server/metadata/db_test.go [hunk 4]
```diff
@@ -203,6 +204,44 @@ func TestOpenDropsPersistedWorkflowEvents(t *testing.T) {
...
+	now := time.Now().UTC().UnixMilli()
+	execSeed(t, db, "project", `INSERT INTO projects (id, display_name, created_at_unix_ms, updated_at_unix_ms) VALUES ('project-system-comment', 'Project', ?, ?)`, now, now)
+	seedWorkflowGraph(t, db, "project-system-comment", now)
+	execSeed(t, db, "workflow task", workflowSeedTaskSQL, "task-system-comment", "link-1", 1, "SYS-1", now, now)
+	execSeed(t, db, "workflow placement", workflowSeedPlacementSQL, "placement-system-comment", "task-system-comment", "node-start", now, now)
+	if _, err := db.Exec(`PRAGMA ignore_check_constraints = ON`); err != nil {
+		t.Fatalf("enable check constraint bypass: %v", err)
+	}
+	execSeed(t, db, "legacy system comment", `INSERT INTO task_comments (id, task_id, body, author_kind, author_id, created_at_unix_ms, updated_at_unix_ms) VALUES ('comment-system', 'task-system-comment', 'legacy system note', 'system', '', ?, ?)`, now, now)
+	if _, err := db.Exec(`PRAGMA ignore_check_constraints = OFF`); err != nil {
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/metadata/db_test.go:235]: The PRAGMA bypass is functionally required, not just a pooling guard: at version 32 task_comments has `CHECK (author_kind IN ('user','agent'))`, so seeding the legacy 'system' row fails without `ignore_check_constraints`. Verified empirically — removing it yields `CHECK constraint failed: author_kind IN ('user','agent')`. Keeping as-is.

File: server/serverstatus/service.go [hunk 5]
```diff
@@ -23,19 +24,19 @@ func NewService(authManager *auth.Manager, cfg config.App) *Service {
...
 		if err != nil {
 			return serverapi.ServerReadinessResponse{}, err
 		}
 		authReady = auth.EvaluateStartupGate(state).Ready
 	}
-	if s != nil {
-		settings = s.settings
-	}
-	ready := authReady
+	ready := authReady || !authRequired
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/serverstatus/service.go:42]: Leaving as-is intentionally. Treating empty/zero settings as 'auth required' would regress the deliberate design that non-OpenAI providers need no startup auth and report ready (this is exactly what the readiness fix established). An empty model is a config-load concern, not a readiness concern, and the `s == nil` path is unreachable in production (the service is always constructed via NewService). The valid half — not failing readiness on the auth store when auth isn't required — is fixed in 0f3d8873.

File: cli/builder/workflow_command_test.go [hunk 6]
```diff
@@ -116,24 +123,49 @@ func TestWorkflowAndTaskCommandsUseWorkflowAPI(t *testing.T) {
 	}
 
 	taskOut, _ := runWorkflowRootCommandOK(t, "task", "create", "--title", "Task", "--body", "Body", "--workflow", workflowID, "--project", binding.ProjectID)
-	taskID := labeledOutputValue(t, taskOut, "task_id")
-	shortID := labeledOutputValue(t, taskOut, "short_id")
-	if taskID == "" || shortID == "" {
-		t.Fatalf("task output = %q, want task and short ids", taskOut)
+	shortID := taskDetailHeadingShortID(t, taskOut)
+	if !strings.Contains(taskOut, shortID+": Task\n") || !strings.Contains(taskOut, "Body:\n```md\nBody\n```\n") || !strings.Contains(taskOut, "Status: open\n") {
+		t.Fatalf("task create output = %q, want task show output", taskOut)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/workflow_command_test.go:128]: Resolving without converting these. The human-readable CLI output is this tool's primary product surface, and each rendered-output check here is already paired with a `--json` assertion that validates structure/parsing. Converting the human-readable checks to JSON would leave the text renderer (a real feature) untested. These assert that task/workflow field values render, not incidental prose, so they keep meaningful signal on the renderer.

File: server/metadata/migrations/00033_remove_system_task_comment_author.up.sql [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    body,
+    CASE WHEN author_kind = 'system' THEN 'agent' ELSE author_kind END,
+    CASE WHEN author_kind = 'system' AND author_id = '' THEN 'system' ELSE author_id END,
+    created_at_unix_ms,
+    updated_at_unix_ms
+FROM task_comments
+ORDER BY rowid ASC;
+
+DROP TABLE task_comments;
+ALTER TABLE task_comments_new RENAME TO task_comments;
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/metadata/migrations/00033_remove_system_task_comment_author.up.sql:37]: Fixed in 098ea383: recreate task_comments_task_updated_idx after the table rename, plus an index-exists assertion.

File: server/workflowsvc/service.go [hunk 8]
```diff
@@ -772,15 +775,40 @@ func (s *Service) ListWorkflowTaskComments(ctx context.Context, req serverapi.Wo
...
 		return serverapi.WorkflowTaskCommentListResponse{}, err
 	}
+	comments, err := s.store.ListCommentsPage(ctx, workflow.TaskID(req.TaskID), offset, pageSize+1)
+	if err != nil {
+		return serverapi.WorkflowTaskCommentListResponse{}, err
+	}
+	nextPageToken := ""
+	if len(comments) > pageSize {
+		comments = comments[:pageSize]
+		nextPageToken = strconv.Itoa(offset + pageSize)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowsvc/service.go]: Fixed in 4f37fa2b: comment pagination now uses a keyset (created_at_unix_ms, id) cursor like activity. Added a store test proving a newer comment inserted between pages can't duplicate/skip.

File: apps/desktop/src/features/task-detail/TaskDetailList.tsx [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+      items={listItems}
+      loadingLabel={t("app.loadingMore")}
+      loadMoreKey={paging.loadMoreKey}
+      onLoadMore={paging.loadMore}
+      rowSpacing="compact"
+      renderItem={(item) => (
+        <TaskDetailListRow
+          activityCount={activityItems.length}
+          commentCount={commentItems.length}
+          detail={detail}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/task-detail/TaskDetailList.tsx:119]: Fixed in eb97b661: comment tab count now uses detail.comments.length (full count from the task detail response) instead of the loaded page length.

File: server/serverstatus/service.go [hunk 10]
```diff
@@ -23,19 +24,19 @@ func NewService(authManager *auth.Manager, cfg config.App) *Service {
 
 func (s *Service) GetServerReadiness(ctx context.Context, _ serverapi.ServerReadinessRequest) (serverapi.ServerReadinessResponse, error) {
 	authReady := false
-	authRequired := true
 	settings := config.Settings{}
+	if s != nil {
+		settings = s.settings
+	}
+	authRequired := authpolicy.RequiresStartupAuth(settings)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/serverstatus/service.go:31]: Fixed in 0f3d8873: readiness only consults the auth store when startup auth is required, so a corrupt/inaccessible auth file can't block a non-OpenAI setup. Added tests.

File: apps/desktop/src/api/schemas/workflow.ts [hunk 11]
```diff
@@ -815,3 +816,13 @@ const taskSummaryResponseSchema = z.object({ task: z.object({ id: z.string() })
 export const taskCreateResponseSchema = taskSummaryResponseSchema;
 export const taskUpdateResponseSchema = taskSummaryResponseSchema;
 export const commentAddResponseSchema = z.object({ comment: commentSchema });
+
+export const commentPageSchema: z.ZodType<CommentPage> = z
+  .object({
+    comments: z.array(commentSchema),
+    next_page_token: z.string().optional().default(""),
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/api/schemas/workflow.ts:823]: Fixed in eb97b661: commentPageSchema.comments now uses .nullish().transform(emptyArray), matching taskDetailSchema.

File: apps/desktop/src/app/sidebarSizing.ts [hunk 12]
```diff
@@ -1,26 +1,60 @@
...
+};
+
+export function clampSidebarWidth(
+  widthPx: number,
+  maxWidthPx = Number.MAX_SAFE_INTEGER,
+  minWidthPx = sidebarMinWidthPx,
+): number {
+  const roundedMaxWidthPx = Math.max(0, Math.round(maxWidthPx));
+  const effectiveMinWidthPx = Math.min(Math.max(0, Math.round(minWidthPx)), roundedMaxWidthPx);
+  return Math.min(Math.max(Math.round(widthPx), effectiveMinWidthPx), roundedMaxWidthPx);
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/app/sidebarSizing.ts]: Fixed in eb97b661: clampSidebarWidth and sidebarResizeBoundsForShellWidth now coerce non-finite values via normalizePx.

File: apps/desktop/src/features/task-detail/TaskDetailContent.tsx [hunk 13]
```diff
@@ -37,35 +16,36 @@ type TaskDraftState = Readonly<{
...
-  const draft = draftState.sourceKey === draftSourceKey ? draftState.draft : taskDraft(detail);
-  const setDraft = (nextDraft: TaskDraft): void => {
-    setDraftState({ sourceKey: draftSourceKey, draft: nextDraft });
-  };
+  const [editingComment, setEditingComment] = useState<Readonly<{ id: string; body: string }> | null>(null);
+  const [newCommentBody, setNewCommentBody] = useState("");
+  const [selectedTab, setSelectedTab] = useState<"comments" | "activity">("comments");
+  const [questionSelections, setQuestionSelections] = useState<ReadonlyMap<string, QuestionSelectionState>>(
+    () => new Map(),
+  );
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/task-detail/TaskDetailContent.tsx:44]: Fixed in eb97b661: editingComment/newCommentBody/questionSelections reset when detail.id changes (render-time adjust-on-prop-change, no effect).

File: apps/desktop/src/features/task-detail/TaskDetailList.tsx [hunk 14]
```diff
@@ -0,0 +1,494 @@
...
+  if (commentsPending) {
+    return [{ kind: "comments-loading" }];
+  }
+  if (commentsError != null) {
+    return [{ kind: "comments-error", error: commentsError }];
+  }
+  if (commentItems.length === 0) {
+    return [{ kind: "comments-empty" }];
+  }
+  return [];
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/task-detail/TaskDetailList.tsx]: Fixed in eb97b661: comment/activity status helpers keep already-loaded rows visible; the full loading/error/empty placeholder is only used when no items have loaded yet.

File: apps/desktop/src/features/task-detail/TaskDetailRows.tsx [hunk 15]
```diff
@@ -0,0 +1,306 @@
...
+            disabled={disabled}
+            key={run.id}
+            onClick={() => {
+              void mutations.interrupt.mutateAsync(run.id);
+            }}
+            variant="secondary"
+          >
+            {t("board.interrupt")} <span className="font-mono">{run.id}</span>
+          </Button>
+        ))}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/task-detail/TaskDetailRows.tsx]: Fixed in eb97b661: interrupt buttons are now gated behind detail.actions.canInterrupt.

File: apps/desktop/src/ui/VirtualizedInfiniteList.tsx [hunk 16]
```diff
@@ -29,14 +33,18 @@ export function VirtualizedInfiniteList<TItem>({
...
+  ariaLabel,
+  rowSpacing = "default",
+  testId,
   paddingEnd = 0,
   paddingStart = 0,
   className,
 }: VirtualizedInfiniteListProps<TItem>) {
   const scrollRef = useRef<HTMLDivElement | null>(null);
-  const lastLoadMoreItemsLengthRef = useRef(-1);
+  const lastLoadMoreKeyRef = useRef("");
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/ui/VirtualizedInfiniteList.tsx:48]: Fixed in eb97b661: a fetch that settles without advancing loadMoreKey now releases the suppression so scrolling can retry. Extracted resolveLoadMore as a pure helper with unit tests.

File: server/metadata/migrations/00033_remove_system_task_comment_author.up.sql [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+PRAGMA foreign_keys = ON;
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/metadata/migrations/00033_remove_system_task_comment_author.up.sql:56]: Fixed in 098ea383: added a pragma_foreign_key_check fail-fast gate after the rename, before re-enabling foreign_keys (same pattern as migration 00020).

File: server/sleepguard/guard.go [hunk 18]
```diff
@@ -30,5 +51,78 @@ func (g *Guard) Release() {
...
+		}
+		g.implLocked().stop()
+		nextGeneration, nextExited, err := g.startLocked()
+		onError := g.onError
+		g.mu.Unlock()
+		if err != nil {
+			if onError != nil {
+				onError(err)
+			}
+			return
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/sleepguard/guard.go:106]: Fixed in ad2a3cae: alwaysController is now command-driven and re-acquires the inhibitor with retry/backoff after a failure, so a single transient restart failure no longer disables always-mode for the process lifetime. Added a regression test.

File: server/workflowsvc/service_test.go [hunk 19]
```diff
@@ -904,6 +913,31 @@ func TestServiceCommentsAndReadModels(t *testing.T) {
...
+	if len(nextCommentPage.Comments) != 1 || nextCommentPage.NextPageToken != "" {
+		t.Fatalf("second comment page = %+v, want one comment without next token", nextCommentPage)
+	}
+	gotPagedCommentIDs := map[string]int{
+		commentPage.Comments[0].ID:     1,
+		nextCommentPage.Comments[0].ID: 1,
+	}
+	if gotPagedCommentIDs[comment.Comment.ID] != 1 || gotPagedCommentIDs[secondComment.Comment.ID] != 1 || len(gotPagedCommentIDs) != 2 {
+		t.Fatalf("paged comment ids = %+v, want both seeded comments exactly once", gotPagedCommentIDs)
+	}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowsvc/service_test.go:940]: Fixed in 4f37fa2b: added coverage for garbage/negative/non-cursor and oversized page tokens in ListWorkflowTaskComments.

File: server/workflowview/service.go [hunk 20]
```diff
@@ -626,6 +662,108 @@ func (s *Service) ListTaskAttention(ctx context.Context, req serverapi.WorkflowT
...
+			nextPageToken = attentionPageToken(items[len(items)-1])
+			break
+		}
+		items = append(items, item)
+	}
+	if !overflowedItems && len(candidates) > pageSize {
+		last := candidates[len(candidates)-1]
+		nextPageToken = attentionCandidatePageToken(last)
+	}
+	return items, nextPageToken, nil
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowview/service.go]: Fixed in 429598c3: attentionItemsPage now keeps advancing the candidate cursor across batches until pageSize valid items are collected or the stream is exhausted. Added a regression test with dropped validation_blocker candidates ahead of a real approval item.

File: server/workflowview/service.go [hunk 21]
```diff
@@ -1240,6 +1378,34 @@ func activityPageToken(item serverapi.WorkflowTaskActivityItem) string {
...
+	return attentionPageCursor{occurredAtUnixMs: occurredAt, itemID: string(decodedID), hasValue: true}, nil
+}
+
+func attentionPageToken(item serverapi.WorkflowAttentionItem) string {
+	return strconv.FormatInt(item.OccurredAtUnixMs, 10) + "|" + base64.RawURLEncoding.EncodeToString([]byte(item.ID))
+}
+
+func attentionCandidatePageToken(item attentionCandidateRow) string {
+	return strconv.FormatInt(item.occurredAtUnixMs, 10) + "|" + base64.RawURLEncoding.EncodeToString([]byte(item.id))
+}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowview/service.go]: Fixed in 429598c3: attention page tokens now encode project_id and parseAttentionPageToken rejects a token whose project_id doesn't match the request.

File: shared/serverapi/workflow_test.go [hunk 22]
```diff
@@ -181,6 +181,12 @@ func TestWorkflowTaskAndCommentRequestValidation(t *testing.T) {
 	if err := (WorkflowTaskCommentAddRequest{TaskID: "task-1", Body: "comment", Author: "user"}).Validate(); err != nil {
 		t.Fatalf("valid comment add rejected: %v", err)
 	}
+	if err := (WorkflowTaskCommentAddRequest{TaskID: "task-1", Body: "comment", Author: "agent"}).Validate(); err != nil {
+		t.Fatalf("valid agent comment add rejected: %v", err)
+	}
+	if err := (WorkflowTaskCommentAddRequest{TaskID: "task-1", Body: "comment", Author: "system"}).Validate(); err == nil || !strings.Contains(err.Error(), "author") {
+		t.Fatalf("system comment author error = %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply shared/serverapi/workflow_test.go]: Fixed in 4f37fa2b: the assertion now checks the WorkflowRequestValidationError Field/Code invariant via errors.As instead of matching error text.

File: shared/serverapi/workflow.go [hunk 23]
```diff
@@ -1568,11 +1576,32 @@ func (r WorkflowTaskQuestionAnswerRequest) Validate() error {
...
+		return err
+	}
+	if r.PageSize < 0 {
+		return workflowRequestError(WorkflowRequestErrorInvalidMode, "page_size", "page_size must be non-negative")
+	}
+	if strings.TrimSpace(r.PageToken) != r.PageToken {
+		return workflowRequestError(WorkflowRequestErrorInvalidMode, "page_token", "page_token must not have leading or trailing whitespace")
+	}
+	return nil
 }
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply shared/serverapi/workflow.go:1612]: Fixed in 4f37fa2b: WorkflowTaskCommentListRequest.Validate now rejects page_size > WorkflowTaskCommentListMaxPageSize (100).

File: server/workflowview/service.go [hunk 24]
```diff
@@ -626,6 +662,108 @@ func (s *Service) ListTaskAttention(ctx context.Context, req serverapi.WorkflowT
...
+	sessionID              string
+	askID                  string
+	taskTransitionID       string
+	interruptionReason     string
+	interruptionDetailJSON string
+	occurredAtUnixMs       int64
+}
+
+func (s *Service) attentionItemsPage(ctx context.Context, projectID string, pageSize int, cursor attentionPageCursor, roleResolver workflow.RoleResolver) ([]serverapi.WorkflowAttentionItem, string, error) {
+	candidates, err := s.attentionItemCandidates(ctx, projectID, "", cursor, pageSize+1)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowview/service.go]: Fixed in 429598c3: attentionItemsPage now keeps advancing the candidate cursor across batches until the page is full or candidates are exhausted, so ListAttention never returns an empty page with a next token. With that, a non-empty token always means a full page, so the home list (even without loadMoreKey) won't stop early; the loadMore retry fix in eb97b661 also covers failed fetches.

File: cli/builder/task_command.go [hunk 25]
```diff
@@ -96,7 +140,12 @@ func taskCreateSubcommand(args []string, stdout io.Writer, stderr io.Writer) int
 		fmt.Fprintln(stderr, err)
 		return 1
 	}
-	fmt.Fprintf(stdout, "task_id\t%s\nshort_id\t%s\nworkflow_id\t%s\n", resp.Task.ID, resp.Task.ShortID, resp.Task.WorkflowID)
+	task, err := getWorkflowTaskByID(context.Background(), remote, resp.Task.ID)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/task_command.go:143]: Intentional, per maintainer decision: a follow-up read/poll failure after the mutation keeps exit 1 so a failed verification is never reported as success. Same decision as the other task-command exit-code threads.

File: cli/builder/task_command.go [hunk 26]
```diff
@@ -556,6 +839,39 @@ func workflowBoardForProject(ctx context.Context, cfg config.App, remote workflo
...
+		return nil, "", "", err
+	}
+	board := resp.Board
+	tasks := workflowTasksForListPage(board, pageSize)
+	return tasks, board.ProjectID, board.NextPageToken, nil
+}
+
+func workflowTasksForListPage(board serverapi.WorkflowBoard, pageSize int) []serverapi.WorkflowTaskSummary {
+	cards := append([]serverapi.WorkflowBoardTaskCard(nil), board.Cards...)
+	if strings.TrimSpace(board.NextPageToken) == "" && len(cards) < pageSize {
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/task_command.go]: Fixed in d387d4e7: workflowTasksForListPage now surfaces the bounded done preview whenever the open stream is exhausted (no next token), including the boundary case where open cards already fill the page. Board pagination only covers open tasks, so this keeps done tasks reachable. Added test coverage.

File: cli/builder/task_command.go [hunk 27]
```diff
@@ -133,40 +182,91 @@ func taskStartSubcommand(args []string, stdout io.Writer, stderr io.Writer) int
...
+
+func taskListStatus(task serverapi.WorkflowTaskSummary) string {
+	if task.CanceledAt != 0 {
+		return "canceled"
+	}
+	if task.Done {
+		return "done"
+	}
+	if len(task.ActiveNodeIDs) > 0 {
+		return "running"
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/builder/task_command.go:294]: Fixed in d387d4e7: task list status is now derived from the board card's authoritative Status.Kind (taskListStatusFromCardStatus) instead of treating any active node as running, so backlog tasks render as 'open'. Added a unit test mapping each status kind plus integration coverage.

File: server/metadata/queries.sql [hunk 28]
```diff
@@ -1479,7 +1651,31 @@ SELECT
...
+WHERE task_id = sqlc.arg(task_id)
+    AND (
+        sqlc.arg(has_cursor) = 0
+        OR created_at_unix_ms < sqlc.arg(cursor_created_at_unix_ms)
+        OR (
+            created_at_unix_ms = sqlc.arg(cursor_created_at_unix_ms)
+            AND id < sqlc.arg(cursor_id)
+        )
+    )
+ORDER BY created_at_unix_ms DESC, id DESC
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/metadata/queries.sql:1677]: Fixed in ac18db8d: added migration 00034 creating task_comments_task_created_idx on (task_id, created_at_unix_ms DESC, id DESC), matching the comment list/keyset-cursor ORDER BY so SQLite seeks instead of sorting the full history per page.

File: server/workflowstore/queries/list_workflows.sql [hunk 29]
```diff
@@ -1,11 +1,31 @@
...
+        description,
+        version,
+        created_at_unix_ms,
+        updated_at_unix_ms,
+        MAX(
+            updated_at_unix_ms,
+            COALESCE((
+                SELECT MAX(task_records.updated_at_unix_ms)
+                FROM task_records
+                WHERE task_records.workflow_id = workflows.id
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/workflowstore/queries/list_workflows.sql:14]: Not needed: task_records is a VIEW (workflow_id comes from project_workflow_links), so it can't be indexed, and the underlying access is already covered. project_workflow_links has project_workflow_links_workflow_idx on (workflow_id) (00005), and tasks has tasks(project_workflow_link_id, updated_at_unix_ms DESC, id DESC) (00011). Together those drive the correlated MAX(updated_at) lookup per workflow without scanning the task table, so the access pattern is already indexed.
